ClothesApp/Application/Services/OrderService.cs

```diff
@@ -0,0 +1,260 @@
+using Application.Dtos.OrderItems;
+using Application.Dtos.Orders;
+using Application.Dtos.OrderTransactions;
+using Application.Exceptions;
+using Application.Interfaces.Repositories;
+using Application.Interfaces.Services;
+using AutoMapper;
+using Domain.Entities;
+using Domain.Enums;
+
+namespace Application.Services;
+
+public class OrderService : IOrderService
+{
+    private readonly IProductsRepository _productsRepository;
+    
+    private readonly IOrderRepository _orderRepository;
+
+    private readonly IAddressService _addressService;
+    
+    private readonly IUserRepository _userRepository;
+
+    private readonly IOrderItemsRepository _orderItemsRepository;
+
+    private readonly IOrderTransactionsRepository _transactionsRepository;
+
+    private readonly IMapper _mapper;
+
+    public OrderService(IOrderRepository orderRepository, IMapper mapper,
+        IOrderTransactionsRepository transactionsRepository, IUserRepository userRepository,
+        IAddressService addressService, IOrderItemsRepository orderItemsRepository,
+        IProductsRepository productsRepository)
+    {
+        _orderRepository = orderRepository;
+        _mapper = mapper;
+        _transactionsRepository = transactionsRepository;
+        _userRepository = userRepository;
+        _addressService = addressService;
+        _orderItemsRepository = orderItemsRepository;
+        _productsRepository = productsRepository;
+    }
+    
+    public async Task<IList<OrderDto>> GetAll()
+    {
+        var orders = await _orderRepository.GetAll();
+        var ordersDto = _mapper.Map<IList<OrderDto>>(orders);
+        return ordersDto;
+    }
+
+    public async Task<OrderDto> GetById(long id)
+    {
+        var exist = await _orderRepository.DoesExist(id);
+
+        if (!exist)
+        {
+            throw new NotFoundException(Messages.OrderNotFound);
+        }
+
+        var order = await _orderRepository.GetById(id);
+        var orderDto = _mapper.Map<OrderDto>(order);
+
+        return orderDto;
+    }
+
+    public async Task<OrderDto> Add(OrderInputDto orderInputDto)
+    {
+        var exist = await _userRepository.DoesExist(orderInputDto.UserId);
+
+        if (!exist)
+        {
+            throw new NotFoundException(Messages.UserNotFound);
+        }
+
+        var belongs = await _addressService.DoesAddressBelongToUser(orderInputDto.AddressId, orderInputDto.UserId);
+
+        if (!belongs)
+        {
+            throw new BusinessRuleException(Messages.AddressUserConstraint);
+        }
+        
+        var order = _mapper.Map<OrderInputDto, Order>(orderInputDto, opt =>
+            opt.AfterMap((_, dest) =>
+            {
+                dest.OrderStatus = OrderStatusType.InReview;
+                dest.CreatedAt = DateTime.Now;
+            }));
+        await _orderRepository.Insert(order);
```

Comment:
Insterting order before all the validations pass is a bad practice because it will be garbage if something goes wrong